stellargraph/layer/cluster_gcn.py

```diff
@@ -0,0 +1,387 @@
+        kernel_regularizer (str): normalization applied to the kernels of graph convolutional layers
+    """
+
+    def __init__(
+        self, layer_sizes, activations, generator, bias=True, dropout=0.0, **kwargs
+    ):
+        if not isinstance(generator, ClusterNodeGenerator):
+            raise TypeError("Generator should be a instance of ClusterNodeGenerator")
+
+        if len(layer_sizes) != len(activations):
+            raise AssertionError(
+                "The number of given layers should be the same as the number of activations."
+                "However given len(layer_sizes): {} vs len(activations): {}".format(
+                    len(layer_sizes), len(activations)
+                )
+            )
+
+        self.layer_sizes = layer_sizes
+        self.activations = activations
+        self.bias = bias
+        self.dropout = dropout
+        self.generator = generator
+        self.support = 1
+
+        # Optional regulariser, etc. for weights and biases
+        self._get_regularisers_from_keywords(kwargs)
+
+        # Initialize a stack of Cluster GCN layers
+        n_layers = len(self.layer_sizes)
+        self._layers = []
+        for ii in range(n_layers):
+            l = self.layer_sizes[ii]
+            a = self.activations[ii]
+            self._layers.append(Dropout(self.dropout))
+            self._layers.append(
+                ClusterGraphConvolution(
+                    l,
+                    activation=a,
+                    use_bias=self.bias,
+                    final_layer=ii == (n_layers - 1),
+                    **self._regularisers,
+                )
+            )
+
+    def _get_regularisers_from_keywords(self, kwargs):
+        regularisers = {}
+        for param_name in [
+            "kernel_initializer",
+            "kernel_regularizer",
+            "kernel_constraint",
+            "bias_initializer",
+            "bias_regularizer",
+            "bias_constraint",
+        ]:
+            param_value = kwargs.pop(param_name, None)
+            if param_value is not None:
+                regularisers[param_name] = param_value
+        self._regularisers = regularisers
+
+    def __call__(self, x):
+        """
+        Apply a stack of Cluster GCN-layers to the inputs.
+        The input tensors are expected to be a list of the following:
+        [
+            Node features shape (1, N, F),
+            Adjacency indices (1, E, 2),
+            Adjacency values (1, E),
+            Output indices (1, O)
+        ]
+        where N is the number of nodes, F the number of input features,
+              E is the number of edges, O the number of output nodes.
+
+        Args:
+            x (Tensor): input tensors
+
+        Returns:
+            Output tensor
+        """
+        x_in, out_indices, *As = x
+
+        Ainput = [Lambda(lambda A: K.squeeze(A, 0))(A) for A in As]
+
+        if len(Ainput) != 1:
```

Comment:
Are there future plans to allow multiple adjacency matrices to be passed in? If not, then this all seems overly elaborate.

Reply:
I've removed it.